domain/src/test/java/org/oppia/android/domain/spotlight/SpotlightStateControllerTest.kt

```diff
@@ -0,0 +1,388 @@
+import org.oppia.android.util.logging.EnableConsoleLog
+import org.oppia.android.util.logging.EnableFileLog
+import org.oppia.android.util.logging.GlobalLogLevel
+import org.oppia.android.util.logging.LogLevel
+import org.oppia.android.util.logging.SyncStatusModule
+import org.oppia.android.util.networking.NetworkConnectionUtilDebugModule
+import org.oppia.android.util.platformparameter.LearnerStudyAnalytics
+import org.oppia.android.util.platformparameter.PlatformParameterValue
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Suppress("SameParameterValue", "FunctionName")
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(application = SpotlightStateControllerTest.TestApplication::class)
+class SpotlightStateControllerTest {
+
+  @Inject
+  lateinit var spotlightStateController: SpotlightStateController
+
+  @Inject
+  lateinit var dataProviderTestMonitor: DataProviderTestMonitor.Factory
+
+  private val profileId = ProfileId.newBuilder().setInternalId(0).build()
+  private val profileId1 = ProfileId.newBuilder().setInternalId(1).build()
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+  }
+
+  @Test
+  fun testMarkSpotlightState_validFeature_notYetMarked_returnsSuccess() {
+    val markSpotlightProvider =
+      spotlightStateController.markSpotlightViewed(profileId, FIRST_CHAPTER)
+    dataProviderTestMonitor.waitForNextSuccessfulResult(markSpotlightProvider)
+  }
+
+  @Test
+  fun testMarkSpotlightState_validFeature_alreadyMarked_returnsSuccess() {
+    markSpotlightSeen(FIRST_CHAPTER)
+    val markSpotlightProvider =
+      spotlightStateController.markSpotlightViewed(profileId, FIRST_CHAPTER)
+    dataProviderTestMonitor.waitForNextSuccessfulResult(markSpotlightProvider)
+  }
+
+  @Test
+  fun testMarkSpotlightView_invalidFeature_returnsFailure() {
+    val invalidFeature = Spotlight.FeatureCase.FEATURE_NOT_SET
+    val markSpotlightProvider =
+      spotlightStateController.markSpotlightViewed(profileId, invalidFeature)
+    dataProviderTestMonitor.waitForNextFailureResult(markSpotlightProvider)
+  }
+
+  @Test
+  fun testMarkSpotlightState_validFeature_differentProfile_returnsSuccess() {
+    val markSpotlightProvider =
+      spotlightStateController.markSpotlightViewed(profileId1, FIRST_CHAPTER)
+    dataProviderTestMonitor.waitForNextSuccessfulResult(markSpotlightProvider)
+  }
+
+  @Test
+  fun testRetrieveSpotlightViewState_firstChapter_notMarked_returnsSpotlightStateNotSeen() {
+    val retrieveSpotlightStateProvider =
+      spotlightStateController.retrieveSpotlightViewState(profileId, FIRST_CHAPTER)
+    val result = dataProviderTestMonitor.waitForNextSuccessfulResult(retrieveSpotlightStateProvider)
+    assertEquals(result, SpotlightViewState.SPOTLIGHT_NOT_SEEN)
+  }
+
+  @Test
+  fun testRetrieveSpotlightViewState_firstChapter_marked_returnsSpotlightStateSeen() {
```

Comment:
Perhaps a test verifying that marking a _different_ spotlight does not cause one to be marked as seen) would also be nice to add.

Reply:
done